modules/juce_audio_devices/native/juce_android_Audio.cpp

```diff
@@ -397,6 +419,23 @@ class AndroidAudioIODevice  : public AudioIODevice,
 
                 env->ReleaseShortArrayElements (audioBuffer, dest, 0);
                 jint numWritten = env->CallIntMethod (outputDevice, AudioTrack.write, audioBuffer, 0, actualBufferSize * numDeviceOutputChannels);
+                int frameCount = numWritten / numDeviceOutputChannels;
+                nextOutputFrameIndex += frameCount;
+                framesSinceOutputLatencyCheck += frameCount;
+
+                // We check for changes in latency every outputLatencyCheckPeriodSec seconds
+                if (framesSinceOutputLatencyCheck >= outputLatencyCheckPeriodSec * outputSampleRate)
+                {
+                    jlong framePosition = 0;
+                    jboolean isValidTimestamp = env->CallBooleanMethod (outputDevice, AudioTrack.getTimestamp, timestamp.get());
+                    if (isValidTimestamp && ((framePosition = env->GetLongField (timestamp, AudioTimestamp.framePosition)) > 0))
+                    {
+                        jlong presentationTime = env->GetLongField (timestamp, AudioTimestamp.nanoTime);
+                        jlong currentTime = env->CallStaticLongMethod(System, System.nanoTime);
+                        calculateAndReportOutputLatency(framePosition, presentationTime, nextOutputFrameIndex, currentTime);

```

Comment:
Here, I would use `std::chrono::nanoseconds` instead of `jlong` as the time type.

Reply:
I wanted to be explicit about the Java data types. But I agree the calculations are then better done with std::chrono constructs.